eslint.config.mjs

```diff
@@ -0,0 +1,186 @@
+      'jsx-a11y/heading-has-content': 'off',
+      'jsx-a11y/anchor-has-content': 'off',
+
+      // This rule doesn't recognise <label> wrapped around custom controls
+      'jsx-a11y/label-has-associated-control': 'off',
+
+      // An overzealous rule that shouts at <a href="#"> in demos.
+      'jsx-a11y/anchor-is-valid': 'off',
+    },
+  },
+  {
+    files: [
+      // matching the pattern of the test runner
+      '**/*.test.?(c|m)[jt]s?(x)',
+    ],
+    extends: createTestConfig(),
+    rules: {
+      'mocha/no-top-level-hooks': 'off', // not needed
+      // disable eslint-plugin-jsx-a11y
+      // tests are not driven by assistive technology
+      // add `jsx-a11y` rules once you encounter them in tests
+      'jsx-a11y/click-events-have-key-events': 'off',
+      'jsx-a11y/control-has-associated-label': 'off',
+      'jsx-a11y/iframe-has-title': 'off',
+      'jsx-a11y/label-has-associated-control': 'off',
+      'jsx-a11y/mouse-events-have-key-events': 'off',
+      'jsx-a11y/no-noninteractive-tabindex': 'off',
+      'jsx-a11y/no-static-element-interactions': 'off',
+      'jsx-a11y/tabindex-no-positive': 'off',
+
+      // In tests this is generally intended.
+      'react/button-has-type': 'off',
+
+      // disable some mocha rules
+      'mocha/no-async-suite': 'off', // not needed
+      'mocha/no-identical-title': 'off', // not needed
+    },
+  },
+  baseSpecRules,
+  {
+    files: ['packages/**/*.test{.tsx,.js}'],
+    rules: {
+      'testing-library/prefer-screen-queries': 'off', // TODO: enable and fix
+      'testing-library/no-container': 'off', // TODO: enable and fix
+      'testing-library/render-result-naming-convention': 'off', // False positives
+    },
+  },
+  {
+    name: 'MUI ESLint config for docs',
+    files: ['docs/**/*'],
+    extends: createDocsConfig(),
+    rules: {
+      'no-restricted-imports': [
+        'error',
+        {
+          paths: NO_RESTRICTED_IMPORTS_PATHS_TOP_LEVEL_PACKAGES,
+          patterns: NO_RESTRICTED_IMPORTS_PATTERNS_DEEPLY_NESTED,
+        },
+      ],
+    },
+  },
+  {
+    files: ['docs/src/app/(private)/experiments/**/*{.tsx,.js}'],
+    rules: {
+      '@typescript-eslint/no-use-before-define': 'off',
+      'no-alert': 'off',
+      'no-console': 'off',
+      'import/no-relative-packages': 'off',
+    },
+  },
+  {
+    files: ['docs/src/app/(public)/(content)/react/utils/use-render/demos/**/*{.tsx,.js}'],
+    rules: {
+      'jsx-a11y/control-has-associated-label': 'off',
+      'react/button-has-type': 'off',
+    },
+  },
+  {
+    files: ['docs/**/*{.ts,.tsx,.js}'],
+    rules: {
+      'no-restricted-imports': [
+        'error',
+        {
+          patterns: NO_RESTRICTED_IMPORTS_PATTERNS_DEEPLY_NESTED,
+        },
+      ],
+      '@typescript-eslint/no-use-before-define': 'off',
+    },
+  },
+  {
+    name: 'Disable image rule for demos',
+    files: [
+      'docs/src/app/(public)/(content)/**/demos/**/*{.tsx,.js}',
```

Comment:
@brijeshb42 I was thinking, perhaps we should standardize on an extension pattern? I know in core we started using `?(c|m)[jt]s?(x)`, which works quite well. How do you feel about exporting a few presets from the code infra package? e.g.

```tsx
export const EXTENSION_JS_ONLY = '?(c|m)js?(x)';
export const EXTENSION_TS_ONLY = '?(c|m)ts?(x)';
export const EXTENSION_TS = '?(c|m)[jt]s?(x)';
export const EXTENSION_DTS = '?.d.(c|m)ts?(x)';
export const EXTENSION_TEST_FILE = `.test.${EXTENSION_TS}`;
// ...
```

Reply:
Added.